File: Brute Force/Selection Sort/code_edu.cpp
#include "algorithm-visualizer.h"
#include <iostream>


const int				ARRAYSIZE = /*(배열 길이 입력)*/
//visualizer{
ChartTracer				chart("ChartTracer");
Array1DTracer			tracer("Array1DTracer");
LogTracer				logger("LogTracer");
Randomize::Array1D<int> array1D(ARRAYSIZE);
int						D[ARRAYSIZE];
int						count = 0;	//무한루프 방지

void InitSeletionSort()
{
	Layout::setRoot(VerticalLayout({ chart,tracer,logger }));
	array1D.fill(D);
	tracer.set(D);
	tracer.chart(chart);
	Tracer::delay();
}
//}

int main()
{
	InitSeletionSort();
	//visualizer{
	logger.print("original array = [");
	for (int i = 0; i < ARRAYSIZE - 1; i++)
	{
		logger.print(std::to_string(D[i]) + " , ");
	}
	logger.println(std::to_string(D[ARRAYSIZE - 1]) + " ] ");
	//}

	for (int i = 0; i < sizeof(D) / sizeof(int); i++) {
		int minJ = i;
		//visualizer{
		tracer.select(i);
		Tracer::delay();
		//}
		for (int j = i + 1; j < sizeof(D) / sizeof(int); j++) {
			//visualizer{
			tracer.select(j);
			Tracer::delay();
			if (count++ > ARRAYSIZE*ARRAYSIZE*2) return;	//무한루프 방지
			//}
			if (/*(알맞은 조건 입력)*/) {
				minJ = j;
				//visualizer{
				tracer.patch(j);
				Tracer::delay();
				tracer.depatch(j);
				//}
			}
			//visualizer{
			tracer.deselect(j);
			//}
		}
		if (/*(알맞은 조건 입력)*/) {
			//visualizer{
			logger.println("swap " + std::to_string(D[i]) + " and " + std::to_string(D[minJ]));
			//}
			int temp = D[i];
			D[i] = D[minJ];
			D[minJ] = temp;
			//visualizer{
			tracer.patch(i, D[i]);
			tracer.patch(minJ, D[minJ]);
			Tracer::delay();
			tracer.depatch(i);
			tracer.depatch(minJ);
			//}
		}
		//visualizer{
		tracer.deselect(i);
		//}
	}
	//visualizer{
	logger.print("sorted array = [");
	for (int i = 0; i < ARRAYSIZE - 1; i++)
	{
		logger.print(std::to_string(D[i]) + " , ");
	}
	logger.println(std::to_string(D[ARRAYSIZE - 1]) + " ] ");
	//}
}

File: Brute Force/Selection Sort/code_edu.java
import org.algorithm_visualizer.*;
import java.util.Arrays;

public class Main {
    //visualizer{
    private static ChartTracer chartTracer = new ChartTracer();
    private static LogTracer logTracer = new LogTracer("Console");
    //}
    private static int sizeOfArray = /*배열 크기 입력*/
    private static int minValue = /*배열 최소값 입력*/
    private static int maxValue = /*배열 최대값 입력*/

    //array[] :sizeOfArray개의 minValue ~ maxValue 사이의 난수 값을 갖는 배열
    private static Integer [] array = (Integer[]) new Randomize.Array1D(sizeOfArray, new Randomize.Integer(minValue, maxValue)).create();

    Main()
    {
        int length = array.length;
        //visualizer{
        Layout.setRoot(new VerticalLayout(new Commander[]{chartTracer, logTracer}));
        logTracer.printf("original array = %s\n", Arrays.toString(array));
        chartTracer.set(array);
        Tracer.delay();
        //} 
        int minIndex;

        for (int i = 0; i < length; i++)
        {
            //visualizer{
            chartTracer.select(i);
            Tracer.delay();
            //}
            minIndex = i;
            for (int j = i + 1; j < length; j++)
            {
                //visualizer{
                chartTracer.select(j);
                Tracer.delay();
                //}
                if(/*알맞은 조건*/) 
                {
                    //visualizer{
                    chartTracer.patch(j, array[j]);
                    Tracer.delay();
                    chartTracer.depatch(j);
                    //}
                    minIndex = /*?*/;
                }
                //visualizer{
                chartTracer.deselect(j);
                //}
            }
            //알맞은 매개변수 입력
            swap(/*?*/,/*?*/, array);
            //visualizer{
            chartTracer.deselect(i);
            //}
        }

        logTracer.printf("sorted array = %s\n", Arrays.toString(array));
    }
    
    void swap(int x, int y, Integer[] array) {
        int temp = /*?*/
        /*?*/ = /*?*/
        /*?*/ = /*?*/
        //visualizer{
        chartTracer.patch(x, array[x]);
        chartTracer.patch(y, array[y]);
        Tracer.delay();
        chartTracer.depatch(x);
        chartTracer.depatch(y);
        //}
    }

    public static void main(String[] args) {
    	new Main();
    }
}